File: baekjoon/용액.cpp
#include <string>
#include <vector>
#include <cstdlib>
#include <iostream>
using namespace std;
vector<long long> v;
int n;

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie();

	cin >> n;
	long long a;
	long long x, y;
	for (int i = 0; i < n; i++) {
		cin >> a;
		v.push_back(a);
	}

	int r = v.size() - 1;
	int l = 0;
	long long min = abs(v[l] + v[r]);
	x = v[l];
	y = v[r];
	while (l<r)
	{
		if (abs(v[l] + v[r]) < min) {
			min = abs(v[l] + v[r]);
			x = v[l];
			y = v[r];
		}
		else {
			if (v[r] + v[l] > 0) {
				r--;
			}
			else if (v[r] + v[l] == 0) {
				x = v[l];
				y = v[r];
				break;
			}
			else {
				l++;
			}
		}
	}
	cout << x << ' ' << y;

}

File: baekjoon/용액.py
n = int(input())
arr = list(map(int, input().split()))
s = 0
e = n-1
min = 2000000002
a=0
b=0
while s<e:
    if abs(arr[s] + arr[e]) <= min:
        min = abs(arr[s] + arr[e])
        a = arr[s]
        b = arr[e]
    if arr[s] + arr[e] > 0:
        e-=1
    else:
        s+=1
print(a, b)
